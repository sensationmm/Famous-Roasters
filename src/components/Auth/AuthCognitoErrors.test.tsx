import { MockedProvider } from '@apollo/client/testing'
import { render, waitFor } from '@testing-library/react'
import React from 'react'

import { AuthCognitoErrors } from '.'

describe('Auth cognito errors custom auth component', () => {
  const snippet = (errorcode?: string) => (
    <form name="testAuthFormCognitoErrors">
      <AuthCognitoErrors errorCode={errorcode} />
    </form>
  )

  it('Renders correctly for no errors', async () => {
    const { container } = render(
      <MockedProvider defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}>
        {snippet(undefined)}
      </MockedProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for LoginGenericException', async () => {
    const { container } = render(
      <MockedProvider defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}>
        {snippet('LoginGenericException')}
      </MockedProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for RegisterGenericException', async () => {
    const { container } = render(
      <MockedProvider defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}>
        {snippet('RegisterGenericException')}
      </MockedProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for ForgotPasswordGenericException', async () => {
    const { container } = render(
      <MockedProvider defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}>
        {snippet('ForgotPasswordGenericException')}
      </MockedProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for UserNotFoundException', async () => {
    const { container } = render(
      <MockedProvider defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}>
        {snippet('UserNotFoundException')}
      </MockedProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for ConfirmUserNotFoundException', async () => {
    const { container } = render(
      <MockedProvider defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}>
        {snippet('ConfirmUserNotFoundException')}
      </MockedProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for ConfirmUserInvalidParameterException', async () => {
    const { container } = render(
      <MockedProvider defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}>
        {snippet('ConfirmUserInvalidParameterException')}
      </MockedProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for UsernameExistsException', async () => {
    const { container } = render(
      <MockedProvider defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}>
        {snippet('UsernameExistsException')}
      </MockedProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for NotAuthorizedException', async () => {
    const { container } = render(
      <MockedProvider defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}>
        {snippet('NotAuthorizedException')}
      </MockedProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for CodeMismatchException', async () => {
    const { container } = render(
      <MockedProvider defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}>
        {snippet('CodeMismatchException')}
      </MockedProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for other error (generic)', async () => {
    const { container } = render(
      <MockedProvider defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}>
        {snippet('whatever')}
      </MockedProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })
})
