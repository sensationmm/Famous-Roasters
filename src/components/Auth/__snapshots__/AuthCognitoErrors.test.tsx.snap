// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth cognito errors custom auth component Renders correctly for CodeMismatchException 1`] = `
<div>
  <form
    name="testAuthFormCognitoErrors"
  >
    <div
      class="bg-brand-grey-whisper p-4 mb-8"
      data-testid="alertInvalidCode"
    >
      <span
        class="text-sm leading-5"
      >
        Verifizierungscode ungültig oder abgelaufen. Bitte versuch mal einen neuen Code anzufragen.
      </span>
    </div>
  </form>
</div>
`;

exports[`Auth cognito errors custom auth component Renders correctly for ConfirmUserInvalidParameterException 1`] = `
<div>
  <form
    name="testAuthFormCognitoErrors"
  >
    <div
      class="bg-brand-grey-whisper p-4 mb-8"
      data-testid="alertRegisterInvalidParam"
    >
      <span
        class="text-sm leading-5"
      >
        E-mail-Adresse ungültig. Versuch eine andere E-mail-Adresse oder
         
        <a
          class="border-b"
          href="/login"
        >
          Anmelden
          .
        </a>
      </span>
    </div>
  </form>
</div>
`;

exports[`Auth cognito errors custom auth component Renders correctly for ConfirmUserNotFoundException 1`] = `
<div>
  <form
    name="testAuthFormCognitoErrors"
  >
    <div
      class="bg-brand-grey-whisper p-4 mb-8"
      data-testid="alertConfirmUserNotFound"
    >
      <span
        class="text-sm leading-5"
      >
        E-mail-Adresse ungültig. Versuch eine andere E-mail-Adresse oder
         
        <a
          class="border-b"
          href="/register"
        >
          Registrieren
          .
        </a>
      </span>
    </div>
  </form>
</div>
`;

exports[`Auth cognito errors custom auth component Renders correctly for ForgotPasswordGenericException 1`] = `
<div>
  <form
    name="testAuthFormCognitoErrors"
  >
    <div
      class="bg-brand-grey-whisper p-4 mb-8"
      data-testid="alertForgotPasswordGeneric"
    >
      <span
        class="text-sm leading-5"
      >
        E-mail-Adresse ungültig. Versuch eine andere E-mail-Adresse oder
         
        <a
          class="border-b"
          href="/register"
        >
          Registrieren
          .
        </a>
      </span>
    </div>
  </form>
</div>
`;

exports[`Auth cognito errors custom auth component Renders correctly for LoginGenericException 1`] = `
<div>
  <form
    name="testAuthFormCognitoErrors"
  >
    <div
      class="bg-brand-grey-whisper p-4 mb-8"
      data-testid="alertLoginGeneric"
    >
      <span
        class="text-sm leading-5"
      >
        E-mail-Adresse oder Passwort ungültig, bitte prüfen oder
         
        <a
          class="border-b"
          href="/register"
        >
          Registrieren
          .
        </a>
      </span>
    </div>
  </form>
</div>
`;

exports[`Auth cognito errors custom auth component Renders correctly for NotAuthorizedException 1`] = `
<div>
  <form
    name="testAuthFormCognitoErrors"
  >
    <div
      class="bg-brand-grey-whisper p-4 mb-8"
      data-testid="alertIncorrectUserNameOrPassword"
    >
      <span
        class="text-sm leading-5"
      >
        Ungültige Daten
      </span>
    </div>
  </form>
</div>
`;

exports[`Auth cognito errors custom auth component Renders correctly for RegisterGenericException 1`] = `
<div>
  <form
    name="testAuthFormCognitoErrors"
  >
    <div
      class="bg-brand-grey-whisper p-4 mb-8"
      data-testid="alertRegisterGeneric"
    >
      <span
        class="text-sm leading-5"
      >
        E-mail-Adresse oder Passwort ungültig, bitte prüfen oder
         
        <a
          class="border-b"
          href="/login"
        >
          Anmelden
          .
        </a>
      </span>
    </div>
  </form>
</div>
`;

exports[`Auth cognito errors custom auth component Renders correctly for UserNotFoundException 1`] = `
<div>
  <form
    name="testAuthFormCognitoErrors"
  >
    <div
      class="bg-brand-grey-whisper p-4 mb-8"
      data-testid="alertIncorrectUserNameOrPassword"
    >
      <span
        class="text-sm leading-5"
      >
        Ungültige Daten
      </span>
    </div>
  </form>
</div>
`;

exports[`Auth cognito errors custom auth component Renders correctly for UsernameExistsException 1`] = `
<div>
  <form
    name="testAuthFormCognitoErrors"
  >
    <div
      class="bg-brand-grey-whisper p-4 mb-8"
      data-testid="alertIncorrectUserNameOrPassword"
    >
      <span
        class="text-sm leading-5"
      >
        Ungültige Daten
      </span>
    </div>
  </form>
</div>
`;

exports[`Auth cognito errors custom auth component Renders correctly for no errors 1`] = `
<div>
  <form
    name="testAuthFormCognitoErrors"
  />
</div>
`;

exports[`Auth cognito errors custom auth component Renders correctly for other error (generic) 1`] = `
<div>
  <form
    name="testAuthFormCognitoErrors"
  >
    <div
      class="bg-brand-grey-whisper p-4 mb-8"
      data-testid="alertGeneric"
    >
      <span
        class="text-sm leading-5"
      >
        Anmeldungsfehler
      </span>
    </div>
  </form>
</div>
`;
